src/modules/src/lighthouse/lighthouse_core.c

```diff
@@ -171,6 +168,40 @@ void lighthouseCoreLedTimer()
   }
 }
 
+static void lighthouseUpdateSystemType() {
+  // Switch to new pulse processor
+  switch(systemType) {
+    case lighthouseBsTypeV1:
+      pulseProcessorProcessPulse = pulseProcessorV1ProcessPulse;
+      break;
+    case lighthouseBsTypeV2:
+      pulseProcessorProcessPulse = pulseProcessorV2ProcessPulse;
+      break;
+    default:
+      // Do nothing if the type is not in range, stay on the previous processor
+      return;
+      break;
```

Comment:
nit: maybe remove the `break;`?  It might trigger unreachable code on some linters?